packages/fast-form-generator-react/src/form/form.utilities.ts

```diff
@@ -290,3 +335,205 @@ export function getBreadcrumbs(
+        const childOption: IChildOptionItem = getChildOptionBySchemaId(subSchemaId, childOptions);
+
+        if (!childOption) {
+            set(
+                mappedData,
+                reactChildrenDataLocation,
+                Object.assign(
+                    { id: subSchemaId },
+                    React.createElement(
+                        React.Fragment,
+                        { key: `${subSchemaId}-${index}` },
+                        subData
+                    )
+                )
+            );
+        } else {
+            set(
+                mappedData,
+                reactChildrenDataLocation,
+                Object.assign(
+                    { id: subSchemaId },
+                    React.createElement(
+                        childOption.component ,
+                        { key: `${subSchemaId}-${index}`, ...subData }
+                    )
+                )
+            );
+        }
+    });
+
+    return mappedData;
+}
+
+/**
+ * Finds the data locations of children
+ */
+export function getDataLocationsOfChildren(schema: any, data: any, childOptions: IChildOptionItem[]): string[] {
+    const dataLocations: string[] = getLocationsFromObject(data);
+    const schemaLocations: string[] = getLocationsFromObject(schema);
+    // get all schema locations from schema
+    const schemaReactChildrenLocations: string[] = getReactChildrenLocationsFromSchema(schema, schemaLocations);
+    // get all schema locations from data locations
+    const schemaLocationsFromDataLocations: string[] = dataLocations.map((dataLocation: string): string => {
+        return mapSchemaLocationFromDataLocation(dataLocation, data, schema);
+    });
+
+    // get all child locations as schema locations
+    const reactChildrenLocationsAsSchemaLocations: string[] = schemaReactChildrenLocations.filter(
+        (schemaReactChildrenLocation: string) => {
+            return !!schemaLocationsFromDataLocations.find((schemaLocationsFromDataLocation: string): boolean => {
+                return schemaReactChildrenLocation === schemaLocationsFromDataLocation;
+            });
+        }
+    );
+
+    const dataLocationsOfChildren: string[] = [];
+
+    // get all child locations as data locations
+    dataLocations.forEach((dataLocation: string) => {
+        if (
+            !!reactChildrenLocationsAsSchemaLocations.find(
+                (reactChildrenLocationsAsSchemaLocation: string) => {
+                    return mapSchemaLocationFromDataLocation(dataLocation, data, schema) === reactChildrenLocationsAsSchemaLocation;
+                })
+            && !Array.isArray(get(data, dataLocation))
+        ) {
+            dataLocationsOfChildren.push(dataLocation);
+        }
+    });
+
+    // for every child location get nested data locations of children
+    dataLocationsOfChildren.forEach((dataLocationOfChildren: string) => {
+        const dataLocation: string = `${dataLocationOfChildren}.props`;
```

Comment:
why does `getSchemaByDataLocation` not accept your `dataLocation` variable? Is this naming accurate?